src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -509,30 +573,45 @@ private async Task<CompilationWithAnalyzersCacheEntry> CreateCompilationWithAnal
         var analyzerMapBuilder = pooledMap.Object;
 
         // This follows what we do in DiagnosticAnalyzerInfoCache.CheckAnalyzerReferenceIdentity
-        using var _ = ArrayBuilder<DiagnosticAnalyzer>.GetInstance(out var analyzerBuilder);
-        foreach (var reference in _project.Solution.AnalyzerReferences.Concat(_project.AnalyzerReferences))
+        using var _1 = ArrayBuilder<DiagnosticAnalyzer>.GetInstance(out var projectAnalyzerBuilder);
+        using var _2 = ArrayBuilder<DiagnosticAnalyzer>.GetInstance(out var hostAnalyzerBuilder);
+        foreach (var reference in _project.Solution.AnalyzerReferences)
         {
```

Comment:
Perhaps the body of the loop can be a local function and AppendAnalyzerMap can be inlined.

Reply:
➡️ This is small enough that I'll leave it for now.